client/src/pages/Budget.page.jsx

```diff
@@ -1,22 +1,152 @@
+import { Button } from '../ui/atoms/Button';
+import { LocalizedDate } from "ui/atoms/LocalizedDate.jsx";
+import { Money } from "ui/atoms/Money.jsx";
+import { CategoryCell } from "ui/molecules/CategoryCell.jsx";
+import { Loader } from "ui/atoms/Loader.jsx";
+import { NoContent } from "ui/atoms/NoContent.jsx";
+import { Error } from "ui/atoms/Error.jsx";
+import AddIcon from '@mui/icons-material/Add';
 
 export const BudgetPage = () => {
+  const [results, setResults] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(false);
+
+  function renderCreatedAt(content) {
+    return LocalizedDate({ date: content.createdAt });
+  }
+
+  function renderCurrentSpending(content) {
+    return <Money inCents={content.currentSpending} />;
+  }
+
+  function renderPlannedSpending(content) {
+    return (<Money inCents={(Math.round(parseFloat(content.currentSpending * 100) / content.currentSpendingPercent))} />);
+  }
+
+  function renderStatus(content) {
+    if (content.currentSpending === content.amountInCents) {
+      return ("Wykorzystany");
+    } else if (content.currentSpending > content.amountInCents) {
+      return ("Przekroczone");
+    } else
+      return ("W normie");
+  }
+
+  function renderName(content) {
+    return (<CategoryCell color={'#37bad7'} name={content.category.name} />);
+  }
+
+  function deleteAction(id) {
+    BudgetService.remove({ ids: id })
+      .then(() => {
+        RetrieveData();
+      }
+      )
+      .catch(() => {
+        setError(true);
+      }
+      );
+  }
+
+  const btnClick = () => {
+    return (<Button variant={'contained'} color={'primary'} startIcon={<AddIcon />}>
+      Zdefiniuj budżet
+    </Button>);
+  }
+
+  function getUniqueId(content) {
+    return content.id;
+  }
+
+  const [headCells] = useState([
+    {
+      id: 1,
+      label: "Nazwa",
+      disablePadding: false,
+      renderCell: renderName
+    }, {
+      id: 2,
+      label: "Planowane wydatki",
+      disablePadding: false,
+      renderCell: renderPlannedSpending
+    }, {
+      id: 3,
+      label: "Obecna kowota",
+      disablePadding: false,
+      renderCell: renderCurrentSpending
+    },
+    {
+      id: 4,
+      label: "Status",
+      disablePadding: false,
+      renderCell: renderStatus
+    },
+    {
+      id: 5,
+      label: "Data utworzenia",
+      disablePadding: false,
+      renderCell: renderCreatedAt
+    }
+  ]);
+
+  async function RetrieveData() {
+    setLoading(true);
+    try {
+      BudgetService.findAll()
+        .then(result => {
+          // po otrzymaniu odpowiedzi - zrob cos z obiektem result
+          result && setResults(result);
+        })
+        .catch(() => {
+          // obsluga erroru
+          setError(true);
+        })
+        .finally(() => {
+          setLoading(false);
+        });
+    } catch (Exception) {
+      setLoading(false);
+      setError(true);
+    }
+  }
```

Comment:
Bardzo ładne wykorzystanie API Promise z małym wyjątkiem. 
- nie jest potrzebny block try catch,API Promise catch & finally jest wystarczające

Zgodnie z konwncją JS nazwy funkcji powinny zaczynać się z małej litery. Wielka litera zarezerwowana jest dla modułów, klas oraz komponentów.

```suggestion
  async function retrieveData() {
    setLoading(true);

      BudgetService.findAll()
        .then(result => {
          // po otrzymaniu odpowiedzi - zrob cos z obiektem result
          result && setResults(result);
        })
        .catch(() => {
          // obsluga erroru
          setError(true);
        })
        .finally(() => {
          setLoading(false);
        });
  }
```